pkg/terraform/attribute.go

```diff
@@ -114,6 +114,18 @@ func (a *Attribute) AsStringValueOrDefault(defaultValue string, parent *Block) t
 	)
 }
 
+func (a *Attribute) AsStringValueSliceOrEmpty(parent *Block) (stringValues []types.StringValue) {
+	if a.IsNil() {
+		return stringValues
+	}
+
+	for _, v := range a.ValueAsStrings() {
+		stringValues = append(stringValues, types.String(v, parent.GetMetadata()))
```

Comment:
I think it should be safe to get the metadata from the attribute itself if it's set:
```suggestion
		stringValues = append(stringValues, types.String(v, a.GetMetadata()))
```